### ISSUE TITLE
Can't archive app due to missing bitcode

### ISSUE THREAD

**Issue by tejassharma96** [DataDog/dd-sdk-swift-testing#55]
### The issue

I'm using this sdk to instrument UI tests, and followed the instructions to add the sdk to my target app. Running the tests works fine, but when I try to archive the app I get the below error:

`ld: '<derived data path>/Build/Intermediates.noindex/ArchiveIntermediates/<app name>/BuildProductsPath/Release-iphoneos/DatadogSDKTesting.framework/DatadogSDKTesting' does not contain bitcode. You must rebuild it with bitcode enabled (Xcode setting ENABLE_BITCODE), obtain an updated library from the vendor, or disable bitcode for this target. file '<derived data path>/Build/Intermediates.noindex/ArchiveIntermediates/<app name>/BuildProductsPath/Release-iphoneos/DatadogSDKTesting.framework/DatadogSDKTesting' for architecture arm64`

Is there some sort of workaround for this? I don't see a way to link the SDK only if I'm running tests. Are you able to publish future versions of the sdk with bitcode enabled?

---

#### Datadog SDK version:

`1.0.0`

#### Last working Datadog SDK version:

n/a

#### Dependency Manager:

SPM and Cocoapods. This sdk is installed using SPM.

#### Other toolset:

No

#### Xcode version:

`Xcode 13.1 (13A1030d)`

#### Swift version:

Whatever ships with Xcode 13.1

#### Deployment Target:

`iOS 13`, `iPhone` + `iPad`

#### macOS version:

`macOS Big Sur 11.6 (20G165)`

**Comment by nachoBonafonte** [comment]
Hi Tejas,
The framework should only be linked with the application when running tests, and you should not distribute the framework to your users. (enabling bitcode is not possible and probably App Store review would not approve some of the low level stuff we are using)
You have several options to link only when testing, for SPM probably the easiest way is linking only in the Debug configuration (or the configuration used to run the tests), you can add the linking commands directly in the build settings, e.g if the framework is located in the SRCROOT path:
```
FRAMEWORK_SEARCH_PATHS=$(FRAMEWORK_SEARCH_PATHS) $(SRCROOT)/DatadogSDKTesting.xcframework/ios-arm64_x86_64-simulator/
LD_RUNPATH_SEARCH_PATHS=$(LD_RUNPATH_SEARCH_PATHS) $(SRCROOT)/DatadogSDKTesting.xcframework/ios-arm64_x86_64-simulator/
OTHER_LDFLAGS=$(OTHER_LDFLAGS) -framework DatadogSDKTesting
```
You can also store this values directly in a `xcconfig` file and set this file in your project test configuration

**Comment by nachoBonafonte** [comment]
There is some more information about it in #45

**Comment by tejassharma96** [comment]
I see, thanks for the info. Will give it a try.

**Comment by tejassharma96** [comment]
Having a very strange issue here. Since it's using SPM I didn't bother updating `LD_RUNPATH_SEARCH_PATHS` or `OTHER_LDFLAGS` since that should be set automatically by xcode (correct me if I'm wrong). This fixed the issue with our archive, but it caused our unit test target to start failing with `Framework not found DatadogSDKTesting`. Previously (ie when I was embedding the sdk into the app target and not into the unit test target at all), running the unit tests would work fine - it only failed trying to archive the app. This is super strange to me.

The first thing I tried as a fix for this was just to embed the sdk into the unit test target using "Link Binary with Libraries", which worked very strangely - it would give me the same error:
```
ld: framework not found DatadogSDKTesting
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```
But then continue to actually run all of the unit tests, and finally exit with code 65 (despite successfully running all the unit tests)

The other thing I tried was to remove the sdk from "Link Binary with Libraries" and use `OTHER_LDFLAGS` for the unit test target too. This brought me back to the initial failure of not finding the framework and exiting immediately (ie without running the tests). This is very strange behaviour and I'm extremely confused so I figured I'd bring it back here to see if you have any ideas. We're running the tests using [fastlane scan](https://docs.fastlane.tools/actions/scan/) with this command:
```
scan(
  scheme: scheme,
  clean: true,
  force_quit_simulator: true,
  code_coverage: true,
  buildlog_path: 'build/test',
  derived_data_path: 'build/test/derivedData',
  suppress_xcode_output: true,
  devices: devices
)
```

**Comment by tejassharma96** [comment]
hmmm another interesting datapoint - adding it to our Podfile as `pod 'DatadogSDKTesting', :configurations => ['Debug']` seems to fix all the issues. We're currently in the middle of transitioning away from cocoapods though, so I'd love any guidance on getting this working with SPM.

**Comment by nachoBonafonte** [comment]
Hi Tejas, 
Yes Cocoapods has good support for build configurations, but is not the case for SPM where you must configure the target yourself. Try with the following steps for SPM:

1. Add the package to the project and link initially with one of the testing targets of your project
<img width="679" alt="Screenshot 2021-11-12 at 13 39 41" src="https://user-images.githubusercontent.com/47814950/141468729-0c618c55-6b47-4e81-90a5-b30748534b2a.png">
2. For all the test targets in your project, check that the framework is linked in the `Build Phases`
<img width="895" alt="Screenshot 2021-11-12 at 13 35 35" src="https://user-images.githubusercontent.com/47814950/141468193-2a00c71e-9844-458d-8f81-4d00446f9ea3.png">
3. And lastly, add `-framework DatadogSDKTesting` only to the `Debug` configuration of the application's `Other linker flags` setting
<img width="896" alt="Screenshot 2021-11-12 at 13 36 40" src="https://user-images.githubusercontent.com/47814950/141468496-cf99a0e3-fae3-4f18-9f5b-d82dce02f1c6.png">
 
Please let me know if it solved the problem for you, or some other issue happened.

**Comment by tejassharma96** [comment]
I think maybe there's some sort of bug with how SPM gathers its packages or something. I tried what you suggested (adding the sdk to all test targets in build phases and to all other targets in ldflags) and ran into the issue I described earlier of getting a "framework not found" error while running tests, but after that error it continues to run all the tests successfully. Might just be due to the complexity of our project 🤷 . I tried just adding it in manually and then using `LD_RUNPATH_SEARCH_PATHS` and `FRAMEWORK_SEARCH_PATHS` on all the targets and that works (with the same ldflags config I was having the SPM issue with).
